packages/sdk/src/evm/core/classes/erc-721.ts

```diff
@@ -167,16 +168,34 @@ export class Erc721<
    * @returns The NFT metadata
    * @twfeature ERC721
    */
-  public async get(tokenId: BigNumberish): Promise<NFT> {
-    const [owner, metadata] = await Promise.all([
-      this.ownerOf(tokenId).catch(() => constants.AddressZero),
-      this.getTokenMetadata(tokenId).catch(() => ({
-        id: tokenId.toString(),
-        uri: "",
-        ...FALLBACK_METADATA,
-      })),
-    ]);
-    return { owner, metadata, type: "ERC721", supply: "1" };
+  public async get<T extends boolean | undefined = undefined>(
+    tokenId: BigNumberish,
+    loadMetadata?: T,
+  ): Promise<T extends true | undefined ? NFT : NFTWithoutMetadata> {
+    if (loadMetadata === false) {
+      const owner = await this.ownerOf(tokenId).catch(
+        () => constants.AddressZero,
+      );
+      const nft: NFTWithoutMetadata = {
+        owner,
+        metadata: {
+          id: tokenId.toString(),
+        },
+        type: "ERC721",
+        supply: "1",
+      };
+      return nft as T extends true | undefined ? NFT : NFTWithoutMetadata;
```

Comment:
I couldn't find a way to solve this without using type assertion.
Googling leads to this open issue: https://github.com/microsoft/TypeScript/issues/33912